### PR TITLE
Rename typeIdentifier to dataType and fix the naming of our constants

### PR DESCRIPTION
**Pull request by chuga**

### CODE REVIEW

File: src/core/UI/RMXOverlayViewController.m [hunk 1]
```diff
@@ -267,23 +267,24 @@ - (void)cellRequestedFullScreenOverlay:(RMXCell *)cell {
...
 - (Class)cellClassForVariable:(RMXVariable *)variable {
-  if (variable.controlType == RMXControlTypeButton) {
+  // TODO(chuga): This is probably not very fast. We should revisit it.
+  if ([variable.controlType isEqualToString:RMXControlTypeButton]) {
     return [RMXCellButton class];
-  } else if (variable.controlType == RMXControlTypeColorList) {
+  } else if ([variable.controlType isEqualToString:RMXControlTypeColorList]) {
     return [RMXCellColorList class];
-  } else if (variable.controlType == RMXControlTypeColorPicker) {
+  } else if ([variable.controlType isEqualToString:RMXControlTypeColorPicker]) {
```
Comment (chriscox) [inline src/core/UI/RMXOverlayViewController.m]: Shouldn't this actually be changed to `RMXControlTypeColorInput`?

File: src/core/UI/RMXOverlayViewController.m [hunk 2]
```diff
@@ -267,23 +267,24 @@ - (void)cellRequestedFullScreenOverlay:(RMXCell *)cell {
...
+  } else if ([variable.controlType isEqualToString:RMXControlTypeSlider]) {
     return [RMXCellSlider class];
-  } else if (variable.controlType == RMXControlTypeStepper) {
+  } else if ([variable.controlType isEqualToString:RMXControlTypeStepper]) {
     return [RMXCellStepper class];
-  } else if (variable.controlType == RMXControlTypeSwitch) {
+  } else if ([variable.controlType isEqualToString:RMXControlTypeSwitch]) {
     return [RMXCellSwitch class];
-  } else if (variable.controlType == RMXControlTypeTextPicker) {
+  } else if ([variable.controlType isEqualToString:RMXControlTypeTextPicker]) {
```
Comment (chriscox) [inline src/core/UI/RMXOverlayViewController.m]: This should be `RMXControlTypeTextList`?

File: src/core/remixes/private/RMXVariableConstants.h [hunk 3]
```diff
@@ -14,34 +14,30 @@
...
+/** Identifiers for the different types of data Variables can hold. */
+static NSString *const RMXDataTypeBoolean = @"__DataTypeBoolean__";
+static NSString *const RMXDataTypeColor = @"__DataTypeColor__";
+static NSString *const RMXDataTypeNumber = @"__DataTypeNumber__";
+static NSString *const RMXDataTypeString = @"__DataTypeString__";
+
+/** Identifiers for the different types of controls Variables can have. */
+static NSString *const RMXControlTypeButton = @"__ControlTypeButton__";
+static NSString *const RMXControlTypeColorList = @"__ControlTypeColorList__";
+static NSString *const RMXControlTypeColorPicker = @"__ControlTypeColorPicker__";
```
Comment (chriscox) [inline src/core/remixes/private/RMXVariableConstants.h]: `static NSString *const RMXControlTypeColorColorInput = @"__ControlTypeColorInput__";`

File: src/core/remixes/private/RMXVariableConstants.h [hunk 4]
```diff
@@ -14,34 +14,30 @@
...
+static NSString *const RMXControlTypeSegmented = @"__ControlTypeSegmented__";
+static NSString *const RMXControlTypeSlider = @"__ControlTypeSlider__";
+static NSString *const RMXControlTypeStepper = @"__ControlTypeStepper__";
+static NSString *const RMXControlTypeSwitch = @"__ControlTypeSwitch__";
+static NSString *const RMXControlTypeTextPicker = @"__ControlTypeTextPicker__";
```
Comment (chriscox) [inline src/core/remixes/private/RMXVariableConstants.h]: static NSString *const RMXControlTypeTextList = @"__ControlTypeTextList__";

**Review by chriscox — CHANGES_REQUESTED** [review]

File: src/core/UI/RMXOverlayViewController.m (hunk 1 above)
Comment (chuga) [reply src/core/UI/RMXOverlayViewController.m]: Done

File: src/core/UI/RMXOverlayViewController.m (hunk 2 above)
Comment (chuga) [reply src/core/UI/RMXOverlayViewController.m]: Done

File: src/core/remixes/private/RMXVariableConstants.h (hunk 3 above)
Comment (chuga) [reply src/core/remixes/private/RMXVariableConstants.h]: Done

File: src/core/remixes/private/RMXVariableConstants.h (hunk 4 above)
Comment (chuga) [reply src/core/remixes/private/RMXVariableConstants.h]: Done

**Review by chriscox — APPROVED** [review]

**Pull request merged**